komapper-processor/src/main/kotlin/org/komapper/processor/EntityMetamodelGenerator.kt

```diff
@@ -150,16 +150,17 @@ internal class EntityMetamodelGenerator(
             val masking = "${p.column.masking}"
             val wrap = when (p.kotlinClass) {
                 is EnumClass -> {
-                    when (p.kotlinClass.strategy) {
-                        EnumStrategy.NAME -> "{ $exteriorTypeName.valueOf(it) }"
-                        EnumStrategy.ORDINAL -> "{ $exteriorTypeName.values()[it] }"
+                    when (val strategy = p.kotlinClass.strategy) {
+                        EnumStrategy.Name -> "{ $exteriorTypeName.valueOf(it) }"
+                        EnumStrategy.Ordinal -> "{ $exteriorTypeName.values()[it] }"
+                        is EnumStrategy.Property -> "{ v -> $exteriorTypeName.values().first { it.${strategy.propertyName} == v }  }"
```

Comment:
Maybe throw some better error if the value isn't found in the enum... it might be a cryptic crash with `first` if there's no clear description of which entity/property/enum and value wasn't found...

Reply:
You are certainly right. I will consider a better error message.

Reply:
Is a null value taken into consideration with this?

Reply:
Yes, of course.
These codes are not called when the value is null.

Reply:
See #817 for better error messages